broker/volumebroker/server/volume_delete.go

```diff
@@ -33,6 +35,19 @@ func (s *Server) DeleteVolume(ctx context.Context, req *ori.DeleteVolumeRequest)
 		return nil, err
 	}
 
+	if encryption := onmetalVolume.Volume.Spec.Encryption; encryption != nil {
```

Comment:
Instead of manually deleting here, we should mark the encryption secret as controlled by the `Volume` it'll be referenced by (see e.g. how we do it with the ignition secret in [`machine_create.go`](https://github.com/onmetal/onmetal-api/blob/2205f7238d9692312dbd8bfe84460293f38b4775/broker/machinebroker/server/machine_create.go#L217)) - then you'll only need to do a `Delete` on the `Volume` and all objects will be automatically garbage collected.